test/server/listener_manager_impl_test.cc

```diff
@@ -2686,30 +2686,12 @@ class OriginalDstTestFilter : public Extensions::ListenerFilters::OriginalDst::O
 };
 
 TEST_F(ListenerManagerImplWithRealFiltersTest, OriginalDstTestFilter) {
-  // Static scope required for the io_handle to be in scope for the lambda below
-  // and for the final check at the end of this test.
-  static int fd;
-  fd = -1;
-  // temporary io_handle to test result of socket creation
-  Network::IoHandlePtr io_handle_tmp = std::make_unique<Network::IoSocketHandleImpl>(0);
-  EXPECT_CALL(*listener_factory_.socket_, ioHandle()).WillOnce(ReturnRef(*io_handle_tmp));
-
   class OriginalDstTestConfigFactory : public Configuration::NamedListenerFilterConfigFactory {
   public:
     // NamedListenerFilterConfigFactory
     Network::ListenerFilterFactoryCb
     createFilterFactoryFromProto(const Protobuf::Message&,
-                                 Configuration::ListenerFactoryContext& context) override {
-      auto option = std::make_unique<Network::MockSocketOption>();
-      EXPECT_CALL(*option, setOption(_, envoy::api::v2::core::SocketOption::STATE_PREBIND))
-          .WillOnce(Return(true));
-      EXPECT_CALL(*option, setOption(_, envoy::api::v2::core::SocketOption::STATE_BOUND))
-          .WillOnce(Invoke(
-              [](Network::Socket& socket, envoy::api::v2::core::SocketOption::SocketState) -> bool {
-                fd = socket.ioHandle().fd();
-                return true;
-              }));
-      context.addListenSocketOption(std::move(option));
```

Comment:
@mattklein123  The tricky test is verifying if `createFilterFactoryFromProto` is called but delegating it to `ListenerFactoryContext::addListenSocketOption`. This might be reasonable in the past but is not appropriate at this moment.

BTW I think 
https://github.com/envoyproxy/envoy/blob/1ca1d13941a14b8fa09ea839d6e936ec32e914b6/source/server/listener_manager_impl.cc#L208-L217 is why OriginalDstListenerFilter(as well as other ListenerFilters) no longer need addListenSocketOption in `ListenerFactoryContext`

Reply:
I'm still concerned that people may have private listener filters that are using this and won't be able to now. Can you see who added the tests that you changed and ask them?